frontend/src/app/shared/dialog/objective-dialog/objective-form.component.ts

```diff
@@ -23,18 +25,22 @@ import { TranslateService } from '@ngx-translate/core';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ObjectiveFormComponent implements OnInit {
+  @ViewChild('alignmentInput') alignmentInput!: ElementRef<HTMLInputElement>;
+
   objectiveForm = new FormGroup({
     title: new FormControl<string>('', [Validators.required, Validators.minLength(2), Validators.maxLength(250)]),
     description: new FormControl<string>('', [Validators.maxLength(4096)]),
     quarter: new FormControl<number>(0, [Validators.required]),
     team: new FormControl<number>({ value: 0, disabled: true }, [Validators.required]),
-    relation: new FormControl<number>({ value: 0, disabled: true }),
+    alignment: new FormControl<AlignmentPossibilityObject | null>(null),
     createKeyResults: new FormControl<boolean>(false),
   });
   quarters$: Observable<Quarter[]> = of([]);
   quarters: Quarter[] = [];
   teams$: Observable<Team[]> = of([]);
-  currentTeam: Subject<Team> = new Subject<Team>();
+  alignmentPossibilities$: Observable<AlignmentPossibility[]> = of([]);
```

Comment:
alignmentPossibilities müsste meiner Meinung nach gar nicht unbedingt ein Observable sein. Du könntest das wohl als normalen Array abbilden, was das ganze etwas einfacher machen würde.
Der Array würde dann in der Funktion generateAlignmentPossibilities abgefüllt und in der Funktion filter() verwendet. Dort könntest du dir dann die Subscriptions, etc. sparen.
Vielleicht übersehe ich aber auch etwas.